devito/finite_differences/differentiable.py

```diff
@@ -379,7 +393,7 @@ def _gather_for_diff(self):
         for f in self.args:
             if f not in self._args_diff:
                 new_args.append(f)
-            elif f is func_args:
+            elif f is func_args or isinstance(f, DifferentiableFunction):
```

Comment:
nitpicking: potentially nicer if you `.extend` the `func_args` with the DifferentiableFunctions upfront?